### PR TITLE
Added api for boards

### PR DESCRIPTION
**Pull request by defnax**

### CODE REVIEW

**Comment by defnax** [comment]
created a clean version of #31

**Comment by defnax** [comment]
@sehraf 
look if this right?
```

	/**
	 * @brief Create a vote
	 * @jsonapi{development}
	 * @param[in]  postGrpIdId  Id of the board where to vote
	 * @param[in]  postMsgId    Id of the board post
	 * @param[in]  voterId      Id of the author that have voted
	 * @return false on error, true otherwise
	 */
	virtual bool voteForPost(bool up,const RsGxsGroupId& postGrpId,const RsGxsMessageId& postMsgId,const RsGxsId& voterId) =0;
```

**Comment by sehraf** [comment]
you missed `bool up`

**Comment by defnax** [comment]
thats right?

```
	/**
	 * @brief Create a vote
	 * @jsonapi{development}
	 * @param[in]  postGrpIdId  Id of the board where to vote
	 * @param[in]  postMsgId    Id of the board post
	 * @param[in]  voterId      Id of the author that have voted
	 * @param[in]  up
	 * @return false on error, true otherwise
	 */
```

**Comment by sehraf** [comment]
the order matters, to `up ` must be first

**Comment by defnax** [comment]
thx its done now

**Comment by defnax** [comment]
@csoler

**Comment by defnax** [comment]
hi what i forget posted class has no subscribe api

this is needed i think for webui and correct?
```
	/**
	 * @brief Request subscription to a group.
	 * @jsonapi{development}
	 * The action is performed asyncronously, so it could fail in a subsequent
	 * phase even after returning true.
	 * @param[out] token Storage for RsTokenService token to track request
	 * status.
	 * @param[in] groupId Board id
	 * @param[in] subscribe
	 * @return false on error, true otherwise
	 */
	virtual bool subscribeToGroup( uint32_t& token, const RsGxsGroupId &groupId,
	                               bool subscribe ) = 0;
```

**Comment by csoler** [comment]
you cannot use the token system for the JSON api. See how channel do it and add the missing function.

**Comment by defnax** [comment]
> you cannot use the token system for the JSON api. See how channel do it and add the missing function.

my first work is done,  @G10h4ck is more expert in this things,  in two weeks i can have a look there

i saw only the code on channels with these which misses for posted

```
/**
* @brief Subscrbe to a channel. Blocking API
* @jsonapi{development}
* @param[in] channelId Channel id
* @param[in] subscribe true to subscribe, false to unsubscribe
* @return false on error, true otherwise
*/
virtual bool subscribeToChannel( const RsGxsGroupId& channelId,
	                                 bool subscribe ) = 0;
```

```
bool p3GxsChannels::subscribeToChannel( const RsGxsGroupId& groupId, bool subscribe )
{
	uint32_t token;
    if( !subscribeToGroup(token, groupId, subscribe) || waitToken(token) != RsTokenService::COMPLETE ) return false;

    RsGxsGroupId grpId;
    acknowledgeGrp(token,grpId);
	return true;
}

```

**Comment by defnax** [comment]
you can merge this? the subscrib part seems more complex that needs to checked/added by @G10h4ck

File: src/retroshare/rsposted.h [hunk 1]
```diff
@@ -213,16 +288,55 @@ class RsPosted : public RsGxsIfaceHelper, public RsGxsCommentService
...
-    virtual void setMessageReadStatus(uint32_t& token, const RsGxsGrpMsgIdPair& msgId, bool read) = 0;
+	/**
+	 * @brief toggle message read status
+	 * @deprecated
+	 * @param[out] token GXS token queue token
+	 * @param[in] msgId
+	 * @param[in] read
+	 */
+	RS_DEPRECATED_FOR(setPostReadStatus)
+	virtual void setMessageReadStatus(uint32_t& token, const RsGxsGrpMsgIdPair& msgId, bool read) = 0;
```
Comment (csoler) [inline src/retroshare/rsposted.h:299]: There shouldn't any JSON API for themethods with token, because they are not compatible.

Comment (defnax) [reply src/retroshare/rsposted.h:299]: i can disable it

Comment (defnax) [reply src/retroshare/rsposted.h:299]: but that is always didabled it i did only what @G10h4ck  did

Comment (defnax) [reply src/retroshare/rsposted.h:299]: i added for readstatus a api but @sehraf said it not needed it yet,

Comment (defnax) [reply src/retroshare/rsposted.h:291]: i has removed for readstatus the api code

**Comment by defnax** [comment]
Hi no feedback more i will be on weekend back

**Comment by defnax** [comment]
@G10h4ck

**Comment by defnax** [comment]
missing api for boards -> createPostV2,  createBoardV2,  subscribeToBoard
```
	/**
	 * @brief Create channel. Blocking API.
	 * @jsonapi{development}
	 * @param[in]  name         Name of the channel
	 * @param[in]  description  Description of the channel
	 * @param[in]  thumbnail    Optional image to show as channel thumbnail.
	 * @param[in]  authorId     Optional id of the author. Leave empty for an
	 *                          anonymous channel.
	 * @param[in]  circleType   Optional visibility rule, default public.
	 * @param[in]  circleId     If the channel is not public specify the id of
	 *                          the circle who can see the channel. Depending on
	 *                          the value you pass for
	 *                          circleType this should be be an external circle
	 *                          if EXTERNAL is passed, a local friend group id
	 *                          if NODES_GROUP is passed, empty otherwise.
	 * @param[out] channelId    Optional storage for the id of the created
	 *                          channel, meaningful only if creations succeeds.
	 * @param[out] errorMessage Optional storage for error messsage, meaningful
	 *                          only if creation fail.
	 * @return False on error, true otherwise.
	 */
	virtual bool createChannelV2(
	        const std::string& name,
	        const std::string& description,
	        const RsGxsImage&  thumbnail = RsGxsImage(),
	        const RsGxsId&     authorId = RsGxsId(),
	        RsGxsCircleType    circleType = RsGxsCircleType::PUBLIC,
	        const RsGxsCircleId& circleId = RsGxsCircleId(),
	        RsGxsGroupId& channelId    = RS_DEFAULT_STORAGE_PARAM(RsGxsGroupId),
	        std::string&  errorMessage = RS_DEFAULT_STORAGE_PARAM(std::string)
	        ) = 0;
```

````
	/**
	 * @brief Subscrbe to a channel. Blocking API
	 * @jsonapi{development}
	 * @param[in] channelId Channel id
	 * @param[in] subscribe true to subscribe, false to unsubscribe
	 * @return false on error, true otherwise
	 */
	virtual bool subscribeToChannel( const RsGxsGroupId& channelId,
	                                 bool subscribe ) = 0;`
```


```
/**
	 * @brief Create channel post. Blocking API.
	 * @jsonapi{development}
	 * @param[in]  channelId    Id of the channel where to put the post. Beware
	 *                          you need publish rights on that channel to post.
	 * @param[in]  title        Title of the post
	 * @param[in]  mBody        Text content of the post
	 * @param[in]  files        Optional list of attached files. These are
	 *                          supposed to be already shared,
	 *                          @see ExtraFileHash() below otherwise.
	 * @param[in]  thumbnail    Optional thumbnail image for the post.
	 * @param[in]  origPostId   If this is supposed to replace an already
	 *                          existent post, the id of the old post. If left
	 *                          blank a new post will be created.
	 * @param[out] postId       Optional storage for the id of the created post,
	 *                          meaningful only on success.
	 * @param[out] errorMessage Optional storage for the error message,
	 *                          meaningful only on failure.
	 * @return false on error, true otherwise
	 */
	virtual bool createPostV2(
	        const RsGxsGroupId& channelId, const std::string& title,
	        const std::string& mBody,
	        const std::list<RsGxsFile>& files = std::list<RsGxsFile>(),
	        const RsGxsImage& thumbnail = RsGxsImage(),
	        const RsGxsMessageId& origPostId = RsGxsMessageId(),
	        RsGxsMessageId& postId = RS_DEFAULT_STORAGE_PARAM(RsGxsMessageId),
	        std::string& errorMessage = RS_DEFAULT_STORAGE_PARAM(std::string) ) = 0;

```

**Comment by defnax** [comment]
done

**Pull request merged**